roles/uperf/tasks/main.yml

```diff
@@ -253,27 +252,22 @@
     - name: set complete to false
       command: "redis-cli set complete false"
 
-    - name: Get client vm status
-      k8s_facts:
-        kind: VirtualMachineInstance
-        api_version: kubevirt.io/v1alpha3
-        namespace: '{{ operator_namespace }}'
-        label_selectors:
-          - app = uperf-bench-client-{{ trunc_uuid }}
-      register: client_vms
-
-    - name: Update resource state
-      operator_sdk.util.k8s_status:
-        api_version: ripsaw.cloudbulldozer.io/v1alpha1
-        kind: Benchmark
-        name: "{{ meta.name }}"
-        namespace: "{{ operator_namespace }}"
-        status:
-          state: Clients Running
-      when: "workload_args.pair|default('1')|int == client_vms | json_query('resources[].status[]')|selectattr('phase','match','Running')|list|length and workload_args.pair|default('1')|int  == (client_vms | json_query('resources[].status.interfaces[0].ipAddress')|length)"
+    - name: Get count of clients ready
+      command: "redis-cli get clients-{{ trunc_uuid }}"
+      register: clients_ready_count
 
     when: resource_kind == "vm"
 
+  - name: Update resource state
+    operator_sdk.util.k8s_status:
+      api_version: ripsaw.cloudbulldozer.io/v1alpha1
+      kind: Benchmark
+      name: "{{ meta.name }}"
+      namespace: "{{ operator_namespace }}"
+      status:
+        state: Clients Running
+    when: "workload_args.pair|default('1')|int == clients_ready_count.stdout|int"
+
```

Comment:
I think this task should be in the previous block as clients_ready_count is only defined when using VMs

Reply:
Good catch. I'll move it up. It would be good for us to do the same with pods, make sure everything is up before triggering start, but I only wanted to change VMs with this PR :)

Reply:
Yeah, makes sense. With pods we don't have to install RPMs so the gap between clients is not that high. But we need to start working in a mechanism to make sure the different samples/test-cases are also synchronized.

Reply:
For sure. I think we could use what I did here with some slight changes for the rest to ensure that syncronization. I also have a ticket to create a VM image for bo so we don't have to do the installs at runtime, save a bunch of time. :)